core/src/main/java/org/jdbi/v3/core/mapper/EnumMapper.java

```diff
@@ -17,23 +17,31 @@
 import java.sql.SQLException;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
 
+import net.jodah.expiringmap.ExpirationPolicy;
+import net.jodah.expiringmap.ExpiringMap;
 import org.jdbi.v3.core.statement.StatementContext;
 
 /**
  * Column mapper for Java {@code enum} types.
  * @param <E> the enum type mapped
  */
 public abstract class EnumMapper<E extends Enum<E>> implements ColumnMapper<E> {
+    private static final ExpiringMap<Class<? extends Enum<?>>, ColumnMapper<?>> NAME_CACHE = ExpiringMap.builder()
+                .expiration(10, TimeUnit.MINUTES)
+                .expirationPolicy(ExpirationPolicy.ACCESSED)
+                .build();
     EnumMapper() {}
 
     /**
      * @param <E> the enum type to map
      * @param type the enum type to map
      * @return an enum mapper that matches on {@link Enum#name()}
      */
+    @SuppressWarnings("unchecked")
     public static <E extends Enum<E>> ColumnMapper<E> byName(Class<E> type) {
-        return new ByName<>(type);
+        return (ColumnMapper<E>) NAME_CACHE.computeIfAbsent(type, e -> new ByName<>(type));
```

Comment:
This could be `ByName::new` because `e == type` but it's all the same

Reply:
I tried that but it didn't compile.  The `Class<E>` is lost; `NAME_CACHE` only knows that the key is an enum class, not `E` in particular.  `ByName::new` is `new ByName<>(key)` rather than `new ByName<>(type)` and then you get an error that `Class<? extends Enum<?>> isn't a subtype of E` or something like that.